Automattic/jetpack
# Add Rating setting to the Video Block when VideoPress is enabled

Fixes 740-gh-Automattic/greenhouse

~~Depends on https://github.com/Automattic/jetpack/pull/17681 for new endpoint to update the rating.~~ merged!

#### Changes proposed in this Pull Request:
* Adds a new `jetpack_videopress` object field to the v2 api for returning videopress data instead of the one-off `jetpack_videopress_guid`
* This PR adds a new sidebar panel for "Video File Settings" (which affects the video file directly, instead of just the current block)
* A setting for 'Rating' is in the new sidebar panel, that auto-saves when the value is changed (same behaviour as Media Library)

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:
phpunit (Tests original field and new field)
`yarn docker:phpunit --filter Test_WPCOM_REST_API_V2_Attachment_VideoPress_`

Manual
1. Apply this PR to your dev environment
2. Run `yarn build-extensions`
3. Ensure you have a paid Jetpack plan that includes video hosting
4. Enable `Enable high-speed, ad-free video player` in Jetpack > Settings > Performance
5. In a post, add a `/video` block and upload a new video
6. Once the video has uploaded, change the value for Rating in the sidebar. It will save automatically.
7. Open the Media Library, find your video and open it
8. The rating in the media library should match the rating you selected on the block
9. Update the rating here to a new value. It will save automatically.
10. Go back to your post and select the video block (if you left the previous window/tab open you may need to refresh the page)
11. The rating should match the rating you selected from the media library
12. View the public page for the post. If you selected a Rating other than `G`, then you should be shown the "Age Gate" before being allowed to play the video.

Editing existing video blocks
1. Select an existing video block and Click the Pencil icon in the toolbar
2. Upload or choose a new video that has a different rating than the current video
3. Observe the Rating in the sidebar, ensure that rating changes to the rating for the new video and that the video preview changes to the new video.

#### Proposed changelog entry for your changes:

*
